torchok/constructor/logger.py

```diff
@@ -1,19 +1,50 @@
+import os
+from argparse import Namespace
 from itertools import chain
 from pathlib import Path
 from typing import Any, Dict, Optional, Union, MutableMapping
 
-from pytorch_lightning.loggers.logger import Logger
+from omegaconf import DictConfig
+from omegaconf.listconfig import ListConfig
 from pytorch_lightning.loggers.csv_logs import CSVLogger
+from pytorch_lightning.loggers.logger import Logger
 from pytorch_lightning.loggers.mlflow import MLFlowLogger
 from pytorch_lightning.loggers.mlflow import rank_zero_only
-from pytorch_lightning.utilities import rank_zero_warn
 from pytorch_lightning.loggers.neptune import NeptuneLogger
 from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
 from pytorch_lightning.loggers.wandb import WandbLogger
+from pytorch_lightning.utilities import rank_zero_warn
 from pytorch_lightning.utilities.logger import _convert_params
 
-from argparse import Namespace
-from omegaconf.listconfig import ListConfig
+
+def create_logger(logger_config: DictConfig) -> Logger:
+    """
+    Create logger based on logger config.
+    """
+    if logger_config is not None:
+        if logger_config.name == 'MLFlowLoggerX':
+            run_name = logger_config.params.run_name
+        else:
+            run_name = logger_config.experiment_name
```

Comment:
Don't call it `run_name` then. Let's have `experimental_name` and an optional `run_name`. Change your paths processing so that the full path is constructed as `log_dir/experimental_name` or `log_dir/experimental_name/run_name` depending on either `run_name` is specified. The current logic is unclear, let's make everything transparent